traits/ctraits.c

```diff
@@ -4070,10 +4090,32 @@ validate_trait_complex(
                 return result;
 
             case 22: /* Callable check: */
-                if (value == Py_None || PyCallable_Check(value)) {
-                    goto done;
+                {
+                    if (PyCallable_Check(value)) {
+                        return value;
+                    }
+                    else if (value == Py_None) {
+
+                        int tuple_size = PyTuple_GET_SIZE(trait->py_validate);
+
+                        //Handle callables without allow_none, default to allow None
+                        if(tuple_size < 2){
+                            Py_INCREF(value);
+                            return value;
+                        }
+
+                        int allow_none = PyObject_IsTrue(PyTuple_GET_ITEM(trait->py_validate, 1));
+
+                        if (allow_none == -1) {
+                            return NULL;
+                        }
+
+                        else if(allow_none) {
```

Comment:
```suggestion
                        else if (allow_none) {
```

Reply:
Made this and all other style changes you requested.